src/ray/rpc/retryable_grpc_client.h

```diff
@@ -0,0 +1,369 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <chrono>
+
+#include "absl/container/btree_map.h"
+#include "absl/strings/str_format.h"
+#include "ray/common/grpc_util.h"
+#include "ray/rpc/client_call.h"
+#include "ray/rpc/grpc_client.h"
+
+namespace ray {
+namespace rpc {
+
+#define INVOKE_RETRYABLE_RPC_CALL(retryable_rpc_client,                       \
```

Comment:
Personal preference: I'm not a fan of macros, it's hard to read and easy to get wrong.